### PR TITLE
Show nodes full name

### PR DESCRIPTION
**Pull request by bbkrr**

### CODE REVIEW

**Review by karlb — COMMENTED** [review]
Causing should be a generic graph analysis tool and not be tied to a specific data set or natural language. Thus, the name mapping should be declared somewhere in the RealRate-Private repository, probably in the industry model definition.

**Comment by bbkrr** [comment]
> Causing should be a generic graph analysis tool and not be tied to a specific data set or natural language. Thus, the name mapping should be declared somewhere in the RealRate-Private repository, probably in the industry model definition.

Okay got it, so the full_name mapping should be done in RealRate-Private project ?

**Comment by karlb** [comment]
> Okay got it, so the full_name mapping should be done in RealRate-Private project ?

Yes, these names relate to the models defined in https://github.com/realrate/RealRate-Private/blob/develop/industry/de_life/model.py#L512-L853 and https://github.com/realrate/RealRate-Private/blob/develop/industry/de_disability/model.py#L69-L429.

I'm not sure if the mapping matches exactly one of these models or both. Do you know, @Holger-Bartel?

**Comment by Holger-Bartel** [comment]
> I'm not sure if the mapping matches exactly one of these models or both. Do you know, @Holger-Bartel?
In this special case the mapping is the same for both models. But in general it is specific to the industry, please take that into account.

**Comment by Holger-Bartel** [comment]
>I'm not sure if the mapping matches exactly one of these models or both. Do you know, @Holger-Bartel?

In this special case the mapping is the same for both models. But in general it is specific to the industry, please take that into account.

**Comment by bbkrr** [comment]
> > I'm not sure if the mapping matches exactly one of these models or both. Do you know, @Holger-Bartel?
> 
> In this special case the mapping is the same for both models. But in general it is specific to the industry, please take that into account.

Okay sure, I will keep that in mind, thanks.

**Comment by bbkrr** [comment]
I have sent a new field `nodes_full_name` from RealRate-Private, so do I also need to change the code in examples of Causing ?
Can you check my recent pull request @karlb

**Comment by bbkrr** [comment]
Since full name mapping will not be sent in Causing, I will continue to do changes accordingly in #15 and this pull request can be canceled and close.

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,10 @@
 /output/*/graphs
 /output/*/logging.txt
 /__pycache__
+
+build/
+causing/__pycache__/
```
Comment (karlb) [inline .gitignore]: We could just add `__pycache__` (without preceding slash) once. I'm sure we want to add any `__pycache__` directories.

File: causing/graph.py [hunk 2]
```diff
@@ -138,11 +139,10 @@ def dot(
...
                 nodeff_str = ""
                 col_str = ""
-            if base_var:  # ToDo: if full_name # yyyy
-                xnode_show = xnode
-                # from apps.life import rr_report
-                # xnode_show = rr_report.full_name(xnode)
-                # xnode_show = rr_report.break_string(xnode_show)
+            if base_var:  # If full_name # yyy
+                xnode_show = (
+                    node_name[str(xnode)] if str(xnode) in node_name else str(xnode)
```
Comment (karlb) [inline causing/graph.py]: Could also be written as
```suggestion
                    node_name.get(str(xnode), str(xnode))
```

**Review by karlb — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (bbkrr) [reply .gitignore]: okay

File: causing/graph.py (hunk 2 above)
Comment (bbkrr) [reply causing/graph.py]: yes, thank you, had almost forgot about this :)

**Pull request merged**